flutter/lib/src/user_interaction/sentry_user_interaction_widget.dart

```diff
@@ -294,68 +294,129 @@ class _SentryUserInteractionWidgetState
+        'Error while handling pointer-up event $event in $SentryUserInteractionWidget',
+        exception: exception,
+        stackTrace: stacktrace,
+      );
+      // ignore: invalid_use_of_internal_member
+      if (_options?.automatedTestMode ?? false) {
+        rethrow;
+      }
     }
   }
 
   void _onTappedAt(Offset position) {
-    final tappedWidget = _getElementAt(position);
-    final keyValue =
-        WidgetUtils.toStringValue(tappedWidget?.element.widget.key);
-    if (tappedWidget == null || keyValue == null) {
+    final tapInfo = _getElementAt(position);
+    if (tapInfo == null) {
       return;
     }
-    final element = tappedWidget.element;
-
-    Map<String, dynamic>? data;
-    // ignore: invalid_use_of_internal_member
-    if ((_options?.sendDefaultPii ?? false) &&
-        tappedWidget.description.isNotEmpty) {
-      data = {};
-      data['label'] = tappedWidget.description;
+
+    final widgetKey = WidgetUtils.toStringValue(tapInfo.element.widget.key);
+    _createBreadcrumbOnTap(tapInfo, widgetKey);
+    _startTransactionOnTap(tapInfo, widgetKey);
+  }
+
+  void _createBreadcrumbOnTap(UserInteractionInfo info, String? widgetKey) {
+    if (!(_options?.enableUserInteractionBreadcrumbs ?? false)) {
+      return;
     }
 
-    const category = 'click';
-    // ignore: invalid_use_of_internal_member
-    if (_options?.enableUserInteractionBreadcrumbs ?? false) {
-      final crumb = Breadcrumb.userInteraction(
-        subCategory: category,
-        viewId: keyValue,
-        viewClass: tappedWidget.type, // to avoid minification
-        data: data,
-      );
-      final hint = Hint.withMap({TypeCheckHint.widget: element.widget});
-      _hub.addBreadcrumb(crumb, hint: hint);
+    final label = _getLabelRecursively(info.element);
+    final data = {
+      'path': _getTouchPath(info.element),
+      if (label != null) 'label': label
+    };
+
+    final crumb = Breadcrumb.userInteraction(
+      subCategory: 'click',
+      viewId: widgetKey,
+      viewClass: info.type, // to avoid minification
+      data: data,
+    );
+    final hint = Hint.withMap({TypeCheckHint.widget: info.element.widget});
+    _hub.addBreadcrumb(crumb, hint: hint);
+  }
+
+  List<Map<String, String?>> _getTouchPath(Element element) {
+    final path = <Map<String, String?>>[];
+
+    bool addToPath(Element element) {
+      // Break at the boundary (i.e. this [SentryUserInteractionWidget]).
+      if (element.widget == widget) {
+        return false;
+      }
+
+      final widgetName = element.widget.runtimeType.toString();
+      if (!widgetName.startsWith('_')) {
+        final info = {
+          'name': WidgetUtils.toStringValue(element.widget.key),
+          'element': _getElementType(element) ?? widgetName,
+          'label': _getLabel(element, true),
+        }..removeWhere((key, value) => value == null);
+        if (info.isNotEmpty) {
+          path.add(info);
+        }
+      }
+
+      return path.length < 10;
```

Comment:
is 10 something we strictly define or is that set by yourself?

Reply:
> is 10 something we strictly define
not that I'm aware of, it's rather arbitrary